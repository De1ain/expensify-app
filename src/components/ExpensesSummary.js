import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = ({ expenseCount, expensesTotal, hiddenExpenseCount }) => {
    const expenseWord = (expenseCount === 1) ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('0,0.00');
    const existWord = (hiddenExpenseCount === 0 || hiddenExpenseCount === 1) ? 'is' : 'are';
    const pluralS = (hiddenExpenseCount === 1) ? '' : 's';

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{expenseCount}</span> {expenseWord} totalling: <span>â‚ª{formattedExpensesTotal}</span>
                </h1>
                <p className="page-header__title">In addition, there {existWord} <span>{hiddenExpenseCount}</span> hidden expense{pluralS}</p>
                <div className="page-header__actions">
                    <Link className="button button--hover-darken" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    const hiddenExpenses = state.expenses.length - visibleExpenses.length;
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses),
        hiddenExpenseCount: hiddenExpenses
    };
};

export default connect(mapStateToProps)(ExpensesSummary);