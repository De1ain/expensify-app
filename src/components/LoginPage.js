import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin, startFacebookLogin, startGithubLogin } from '../actions/auth';

export const LoginPage = ({ startGoogleLogin, startGithubLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control</p>
            <button className="button button--shadow button--login" onClick={startGoogleLogin}>
                <img src="/images/google-icon.jpeg" className="button--icon" />
                <span>Login with Google</span>
            </button>
            <button className="button button--shadow button--login" onClick={startFacebookLogin} disabled>
                <img src="/images/facebook-icon.jpg" className="button--icon" />
                <span>Login with Facebook</span>
            </button> 
            <button className="button button--shadow button--login" onClick={startGithubLogin}>
                <img src="/images/github-icon.png" className="button--icon" />
                <span>Login with GitHub</span>
            </button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startGithubLogin: () => dispatch(startGithubLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);