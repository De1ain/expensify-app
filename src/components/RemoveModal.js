import React from 'react';
import Modal from 'react-modal';

const RemoveModal = (props) => (
    <Modal
        isOpen={!!props.removeButtonClicked}
        // isOpen={true}
        onRequestClose={props.handleRemoveNo}
        contentLabel="RemoveExpense"
        ariaHideApp={false}
        closeTimeoutMS={200}
        className="modal"
        appElement={app}
    >
        <h3 className="modal__title">Remove Expense</h3>
        <p>Are you sure you want to remove expense?</p>
        <button className="button" onClick={props.handleRemoveYes}>Yes</button>
        <button className="button" onClick={props.handleRemoveNo}>No</button>
    </Modal>
);

export default RemoveModal;