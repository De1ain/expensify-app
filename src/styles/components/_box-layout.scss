.box-layout {
    align-items: center;
    background: url("/images/bg_sky.jpg");
    // background: url("/images/bg_turbines.jpg");
    // background: url("/images/bg_night-sea.jpeg");
    background-size: cover;
    background-position: center;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background: fade-out(white, 0.14);
    border-radius: 4px;
    padding: $l-size $m-size;
    text-align: center;
    width: 35rem;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}


