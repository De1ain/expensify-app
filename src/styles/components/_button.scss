.button {
    background: $blue;
    border: none;
    border-radius: 3px;
    color: white;
    display: inline;
    font-weight: 300;
    font-size: $font-size-large;
    line-height: 4rem;
    margin: 1rem;
    padding: $s-size;
    text-align: center;
    text-decoration: none;
    &:disabled {
        background-color: lighten($grey, 40%);
        color: lighten($dark-grey, 10%);
        box-shadow: none;
        opacity: 0.7;
    }
}

.button--hover-darken {
    // transition: background 0.3s ease;
    // transition: opacity 200ms ease-in-out;
    &:hover {
        background-image: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3));
        // background: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3));
        // background: opacity(0.5);
        // background: darken($blue, 10%);
    }
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}

.button--icon {
    float: left;
    height: 4rem;
    width: 4rem;
    margin-right: 1rem;
}

.button--shadow {
    box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.button--login {
    width: 25rem;
}

.button--confirm {
    margin: 2rem;
}

.button--google {
    background: white;
    color: $dark-grey;
}

.button--facebook {
    background: $facebook-blue;
    color: $off-white;
}

.button--github {
    background: $github-black;
    color: $off-white;
}

.dark-bg {
    background-blend-mode: multiply;
    background-color: black;
    width: 100px;
    padding: 5px;
}
