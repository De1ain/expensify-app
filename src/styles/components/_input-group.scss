.input-group {
    display: flex;
    flex-direction: column;
    justify-content: left;
    margin-bottom: $m-size;
    @media (min-width: $desktop-breakpoint) {
        // for large screens
        flex-direction: row;
        margin-bottom: $l-size;
    }
}

.input-group__item {
    margin-bottom: $s-size;
    @media (min-width: $desktop-breakpoint) {
        // for large screens
        margin: 0 $s-size 0 0;
    }
}
