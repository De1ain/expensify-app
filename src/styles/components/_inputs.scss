.text-input {
    border: 1px solid $gray-border-color;
    border-radius: 2px;
    height: 48px;
    font-size: $font-size-large;
    font-weight: 300;
    padding: $s-size;
}

.select {
    @extend .text-input;
    background-color: white;
}

.text-area{
    @extend .text-input;
    height: 10rem;
}