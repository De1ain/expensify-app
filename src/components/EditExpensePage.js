import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import RemoveModal from './RemoveModal';

export class EditExpensePage extends React.Component {
    state = {
        removeButtonClicked: undefined
    };

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    onRemove = () => {
        this.setState(() => ({ removeButtonClicked: true }));

    };

    handleRemoveConfirm = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    handleRemoveCancel = () => {
        this.setState(() => ({ removeButtonClicked: undefined }));
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary button--hover-darken" onClick={this.onRemove}>Remove Expense</button>
                    <RemoveModal
                        removeButtonClicked={this.state.removeButtonClicked}
                        handleRemoveConfirm={this.handleRemoveConfirm}
                        handleRemoveCancel={this.handleRemoveCancel}
                    />
                </div>
            </div>
        );
    };
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id),
        removeButtonClicked: state.removeButtonClicked
    };
};

const mapDispatchToProps = (dispatch, props) => (
    {
        startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
    }
);

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);