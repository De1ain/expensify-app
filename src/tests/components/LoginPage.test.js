import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('Should correctly render login page', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('Should call starGoogletLogin on button click', () => {
    const startGoogleLogin = jest.fn();
    const wrapper = shallow(<LoginPage startGoogleLogin={startGoogleLogin} />);
    wrapper.find('button').at(0).simulate('click');
    expect(startGoogleLogin).toHaveBeenCalled();
});

test('Should call startGithubLogin on button click', () => {
    const startGithubLogin = jest.fn();
    const wrapper = shallow(<LoginPage startGoogleLogin={startGithubLogin} />);
    // console.log(wrapper.debug()); // for debugging purposes
    wrapper.find('button').at(0).simulate('click');
    expect(startGithubLogin).toHaveBeenCalled();
});