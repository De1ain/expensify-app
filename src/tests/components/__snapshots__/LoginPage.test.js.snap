// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render login page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control
    </p>
    <button
      className="button button--shadow button--login"
    >
      <img
        className="button--icon"
        src="/images/google-icon.jpeg"
      />
      <span>
        Login with Google
      </span>
    </button>
    <button
      className="button button--shadow button--login"
      disabled={true}
      onClick={[Function]}
    >
      <img
        className="button--icon"
        src="/images/facebook-icon.jpg"
      />
      <span>
        Login with Facebook
      </span>
    </button>
    <button
      className="button button--shadow button--login"
    >
      <img
        className="button--icon"
        src="/images/github-icon.png"
      />
      <span>
        Login with GitHub
      </span>
    </button>
  </div>
</div>
`;
